tools/c7n_tencentcloud/c7n_tencentcloud/resources/cos.py

```diff
@@ -0,0 +1,306 @@
+# Copyright The Cloud Custodian Authors.
+# SPDX-License-Identifier: Apache-2.0
+
+import json
+import logging
+
+import jmespath
+from qcloud_cos import CosS3Client, CosConfig, CosServiceError
+
+from c7n.filters import Filter, ValueFilter
+from c7n_tencentcloud.provider import resources
+from c7n_tencentcloud.query import QueryResourceManager, ResourceTypeInfo, DescribeSource
+from c7n.utils import type_schema, format_string_values
+from tencentcloud.common import credential
+
+log = logging.getLogger('custodian.cos')
+
+
+class DescribeCos(DescribeSource):
+
+    @staticmethod
+    def get_cos_client(region):
+        cred = credential.DefaultCredentialProvider().get_credentials()
+        config = CosConfig(Region=region, SecretId=cred.secret_id,
+                           SecretKey=cred.secret_key)
+        return CosS3Client(config)
+
+    def resources(self, params=None):
+        resp = self.get_cos_client(self.resource_manager.config.region).list_buckets()
+        action, jsonpath, extra_params = self.resource_type.enum_spec
+        resources = jmespath.search(jsonpath, resp)
+        resources = [r for r in resources if r["Location"] == self.resource_manager.config.region]
+
+        self.augment(resources)
+        return resources
+
+    def get_resource_qcs(self, resources):
+        """
+        get_resource_qcs
+        resource description https://cloud.tencent.com/document/product/598/10606
+        """
+        qcs_list = []
+        for r in resources:
+            # get appid
+            names = r[self.resource_type.id].split('-')
+            appid = names[len(names) - 1]
+
+            qcs = "qcs::{}:{}:uid/{}:{}".format(
+                self.resource_type.service,
+                self.region,
+                appid,
+                r[self.resource_type.id])
+            qcs_list.append(qcs)
+        return qcs_list
+
+
+@resources.register("cos")
+class COS(QueryResourceManager):
```

Comment:
cam we add a doc string here to explain that cos is object storage?

Reply:
done

I have synchronized the master, the code of c7n_kube has a conflict, please help to solve it, thank you